bin/public/installESpower.py

```diff
@@ -0,0 +1,143 @@
+import os
+import iptables
+
+# The version of this module, used to prevent the same script version to be
+# executed more then once on the same host.
+SCRIPT_VERSION = 1
+CONF_SOURCE='/opt/syco/usr/syco-private/var/'
+
+def build_commands(commands):
+    commands.add("install-espower", install_espower, help="Install power modules for elastcisearch")
+    commands.add("uninstall-espower", uninstall_espower, help="Uninstall the power modules for elastic search")
+
+
+def install_espower(args):
+	'''
+	Installation of Elastic search passing rule
+	
+	'''
+	install_rabbit()
+	install_logstash()
+	config_rabbitmq()
+	config_logstash()
+	# Adding iptables rules
+	iptables.add_rabbitmq_chain()
+	iptables.save()
+	
+
+	print("Go to http://ip-address:15672 for rabbit mq ")
+
+def uninstall_espower(args):
+	x('yum remove rabbitmq-server -y')
+	#x('yum remove erlang -y')
+	x('rm -rf /opt/logstash')
+	x('rm -rf /etc/logstash')
+	x('rm -rf /etc/rabbitmq')
+	x('rm -rf /etc/init.d/shipper')
+	x('rm -rf /etc/init.d/index') 
+
+
+
+
+def install_rabbit():
+	'''
+	Installa and setup the rabbit mq server.
+	'''
+
+
+	#check if epel is install else installed
+	if not os.path.isfile("yum install rabbitmq-server -y/etc/yum.repos.d/epel.repo"):
+		x('yum localinstall http://download.fedoraproject.org/pub/epel/6/i386/epel-release-6-8.noarch.rpm -y')
+	x('yum install erlang -y')
+	x('rpm --import http://www.rabbitmq.com/rabbitmq-signing-key-public.asc')
+	x('yum install rabbitmq-server -y')
+
+
+
+def install_logstash():
+    '''
+    Download and install logstash
+    '''
+    x("curl -O  https://download.elasticsearch.org/logstash/logstash/logstash-1.4.2.tar.gz")
+    x('mv logstash-1.4.2.tar.gz /opt/logstash.tar.gz')
+    x('tar -zxvf /opt/logstash.tar.gz -C /opt/')
+    x('rm /opt/logstash.tar.gz')
+    x('rm -rf /opt/logstash')
+    x('ln -s /opt/logstash-1* /opt/logstash')
+
+
+
+def config_logstash():
+	'''
+	There are now defult config for logstash
+	Copy config from
+	1. First from syco-private
+	2. syco var defult config
+	''' 
+	x('cp -r %slogstash /etc/' %CONF_SOURCE)
+	x('chown logstash:logstash -R /opt/logstash')
+	x('cp /opt/syco/usr/syco-private/var/logstash/start/shipper /etc/init.d/')
+	x('cp /opt/syco/usr/syco-private/var/logstash/start/index /etc/init.d/')
+	x('chmod 700 /etc/init.d/shipper')
+	x('chmod 700 /etc/init.d/index')
+	x('chkconfig --add shipper')
+	x('chkconfig --add index')
+	x('chkconfig shipper on')
+	x('chkconfig index on')
+	x('/etc/init.d/shipper start')
+	x('/etc/init.d/index start')
+
+def config_rabbitmq():
+	'''
+	There are now defult config for logstash
+	Copy config from
+	1. First from syco-private
+	2. syco var defult config
+	'''
+	#Remove old certs
+	x('rm -rf /etc/rabbitmq/ssl')
+
+	x('mkdir -p /etc/rabbitmq/ssl/private')
+	x('openssl req -x509 -config /opt/syco/usr/syco-private/var/rabbitmq/ssl/openssl.cnf -newkey rsa:4096 -days 3650 -out /etc/rabbitmq/ssl/cacert.pem -outform PEM -subj /CN=RabbitMQ/ -nodes')
```

Comment:
Same story on syco-private... Since this is probably private there should be an example file in the usr/mod-template/...